### PR TITLE
Assign sort number when creating field(s)

### LINKED ISSUE #18273: "sort" field on the Field entity defaults to null unless fields are reordered.
### Describe the Bug

When creating fields in a collection, the "sort" value defaults to null unless the user does a reorder action. The Directus studio app must have an alternative sorting method which can cope with this, however this is causing inconsistencies when using the schema migration tool. 

https://user-images.githubusercontent.com/21194068/233403795-b152d588-b7dc-412e-b9d7-821c95db430a.mov

### To Reproduce

Create a a new collection. eg "mycollection"
Create 2 of any type of field in the collection
Make an api query to /fields/mycollection

Note that in response, the 'sort' property is not set.
In the directus app, reorder the fields,
Rerun the API query /fields/mycollection
Note that the "sort" property is now set for all the fields in this collection.

### Directus Version

9.25.1

### Hosting Strategy

Directus Cloud

### PR DESCRIPTION
**Pull request by azrikahar**
Fixes #18273, fixes ENG-913

When creating a new field(s) via the api or schema migration, the sort value is null by default. This PR adds logic to:

- Assign sort values when a new collection is created with fields (eg. when creating a collection with system fields via the app)
- Assign and update sort values when a new field is added to a collection and accounts for groups (fields within each group has sort value restarting from 1)

### CODE REVIEW

File: api/src/services/collections.ts [hunk 1]
```diff
@@ -145,7 +145,24 @@ export class CollectionsService {
...
+						);
+
+						for (const [_group, fields] of Object.entries(fieldsWithGroups)) {
+							sortedFieldPayloads = sortedFieldPayloads.concat(
+								fields.map((field, index) => merge({ sort: index + 1 }, field))
+							);
+						}
+					}
+
+					await fieldItemsService.createMany(sortedFieldPayloads, {
```
Comment (jaads) [inline api/src/services/collections.ts:165]: this logic looks actually complicated enough for me to extract it to a separate function and write unit tests for it 🤔 what do you think @azrikahar @rijkvanzanten

**Review by jaads — COMMENTED** [review]
Confirming that the sort value now gets set properly. Both for system fields on creation of a collection and for fields which get added afterwards 🎉

File: api/src/services/collections.ts (hunk 1 above)
Comment (paescuj) [reply api/src/services/collections.ts:165]: Yeah, I think a unit test would be appropriate here 👍

Comment (jaads) [reply api/src/services/collections.ts:165]: as discussed, no unit tests in this PR

**Review by jaads — APPROVED** [review]

File: api/src/services/collections.ts (hunk 1 above)
Comment (paescuj) [reply api/src/services/collections.ts:165]: > as discussed, no unit tests in this PR

In place of a unit test here, @azrikahar would you be able to add a few explaining comments to make sure we'll understand the logic at a later point?

File: .changeset/dull-kings-marry.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@directus/api': patch
+---
+
+added sort number when creating fields
```
Comment (paescuj) [inline .changeset/dull-kings-marry.md:5]:
```suggestion
Implemented sort number for fields in collections on creation to ensure order is always retained
```

File: api/src/services/collections.ts (hunk 1 above)
Comment (azrikahar) [reply api/src/services/collections.ts:171]: > In place of a unit test here, @azrikahar would you be able to add a few explaining comments to make sure we'll understand the logic at a later point?

@paescuj added comments to clarify the logic used here 👍

Comment (paescuj) [reply api/src/services/collections.ts:171]: Perfect! ❤️

**Review by paescuj — APPROVED** [review]

**Pull request merged**